Structure_Engine/Create/Loads/BarDifferentialTemperatureLoad.cs

```diff
@@ -0,0 +1,114 @@
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Geometry;
+using BH.oM.Structure.Loads;
+using BH.oM.Base;
+using BH.oM.Structure.Elements;
+using System.Collections.Generic;
+using System.Linq;
+using System;
+using System.ComponentModel;
+using BH.oM.Reflection.Attributes;
+
+namespace BH.Engine.Structure
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Creates an differential temprature load to be applied to Bars.")]
+        [InputFromProperty("loadcase")]
+        [Input("positions", "The parametric distance distance from the top (local z) or left (local y) of the profile.")]
+        [Input("temperatures", "The temperature at the corresponding position on the profile.")]
+        [Input("localDirection", "The local direction of the temperature variation relative to the profile. Typically limited to local y or z.")]
+        [Input("objects", "The collection of Bars the load should be applied to.")]
+        [Input("name", "The name of the created load.")]
+        [Output("barDiffTempLoad", "The created BarDifferentialTemperatureLoad.")]
+        public static BarDifferentialTemperatureLoad BarDifferentialTemperatureLoad(Loadcase loadcase, List<double> positions, List<double> temperatures, DifferentialTemperatureLoadDirection localLoadDirection, IEnumerable<Bar> objects, string name = "")
+        {
+            //Checks for positions and profiles
+            if (positions.Count != temperatures.Count)
+            {
+                Reflection.Compute.RecordError("Number of positions and temperatures provided are not equal");
+                return null;
+            }
+            else if (positions.Exists((double d) => { return d > 1; }) || positions.Exists((double d) => { return d < 0; }))
+            {
+                Reflection.Compute.RecordError("Positions must exist between 0 and 1 (inclusive)");
+                return null;
+            }
+
+            // Check that top and bottom positions are included
+            if(!(positions.Contains(0) && positions.Contains(1)))
+            {
+                Reflection.Compute.RecordError("Positions must inlude the bottom (0) and top (1) position.");
+                return null;
+            }
+
+            List<double> sortedPositions = positions;
+            sortedPositions.Sort();
+
+            if (!positions.SequenceEqual(sortedPositions))
+            {
+                Reflection.Compute.RecordError("Positions must be sorted in ascending order.");
+                return null;
+            }
+
+            //Create ditionary for TaperedProfile
+            Dictionary<double, double> temperatureProfile = positions.Zip(temperatures, (z, T) => new { z, T })
```

Comment:
@FraserGreenroyd duplicate and incorrect error. Unless it's referring to the double?